src/lib/DocSearch.js

```diff
@@ -79,38 +79,44 @@ class DocSearch {
       this.autocompleteOptions.cssClasses.prefix || 'ds';
 
     // eslint-disable-next-line no-param-reassign
-    handleSelected = handleSelected || this.handleSelected;
-
     this.isSimpleLayout = layout === 'simple';
 
     this.client = algoliasearch(this.appId, this.apiKey);
     this.client.addAlgoliaAgent(`docsearch.js ${version}`);
+    let uniqInput = null;
+    console.log(this.input);
+    for (uniqInput in this.input) {
+      if (uniqInput) {
+        uniqInput = this.input[uniqInput];
+        if (enhancedSearchInput) {
+          uniqInput = DocSearch.injectSearchBox(uniqInput);
+        }
 
-    if (enhancedSearchInput) {
-      this.input = DocSearch.injectSearchBox(this.input);
-    }
-
-    this.autocomplete = autocomplete(this.input, autocompleteOptions, [
-      {
-        source: this.getAutocompleteSource(transformData, queryHook),
-        templates: {
-          suggestion: DocSearch.getSuggestionTemplate(this.isSimpleLayout),
-          footer: templates.footer,
-          empty: DocSearch.getEmptyTemplate(),
-        },
-      },
-    ]);
-    this.autocomplete.on(
-      'autocomplete:selected',
-      handleSelected.bind(null, this.autocomplete.autocomplete)
-    );
-    this.autocomplete.on(
-      'autocomplete:shown',
-      this.handleShown.bind(null, this.input)
-    );
+        this.autocomplete = autocomplete(uniqInput, autocompleteOptions, [
+          {
+            source: this.getAutocompleteSource(transformData, queryHook),
+            templates: {
+              suggestion: DocSearch.getSuggestionTemplate(this.isSimpleLayout),
+              footer: templates.footer,
+              empty: DocSearch.getEmptyTemplate(),
+            },
+          },
+        ]);
+        if (handleSelected) {
```

Comment:
Yes this is why this one is only here to ensure backward compatibility. It enables method that were overridden

Reply:
Did you double check that it was still working? Because we now have links instead of divs, it's possible the default `click` event from the browser (moving to the next page) is triggered before the custom `handleSelected` and thus `handleSelected` will never be triggered.

If `handleSelected` is custom-defined, we should make sure the default event on the link is captured and redirected to `handleSelected`

Reply:
No I haven't run any test so far except human emulation